app/api/datastore/entities/search_result.rb

```diff
@@ -0,0 +1,24 @@
+module Datastore
+  module Entities
+    class SearchResult < Grape::Entity
+      expose :id, as: :resource_id
+      expose :submitted_at
+      expose :applicant_name
+      expose :reference
+
+      private
+
+      def applicant_name
+        "#{applicant&.dig('first_name')} #{applicant&.dig('last_name')}"
```

Comment:
This really smells like DynamoDB all over again 😄 

In DynamoDB the idea was to have hash keys, `first_name#last_name#dob` etc.

As you advance, keep an eye to see if in reality everything that we are doing (apart from pagination) with Postgres might not be better suited with Dynamo 👍 
Pagination in an extreme situation I think could be handled as well...